packages/nimble-components/src/table/models/keyboard-navigation-manager.ts

```diff
@@ -0,0 +1,1370 @@
+            if (row) {
+                if (!(row instanceof TableGroupRow)) {
+                    const cell = this.getContainingCell(activeElement);
+                    if (cell) {
+                        const columnIndex = this.table.visibleColumns.indexOf(
+                            cell.column!
+                        );
+                        if (cell.actionMenuButton === activeElement) {
+                            this.setCellActionMenuFocusState(
+                                row.resolvedRowIndex!,
+                                columnIndex,
+                                false
+                            );
+                            return;
+                        }
+                        const contentIndex = cell.cellView.tabbableChildren.indexOf(
+                            activeElement
+                        );
+                        if (contentIndex > -1) {
+                            this.setCellContentFocusState(
+                                contentIndex,
+                                row.resolvedRowIndex!,
+                                columnIndex,
+                                false
+                            );
+                            return;
+                        }
+                    }
+                }
+                if (
+                    setRowFocus
+                    && this.hasRowOrCellFocusType()
+                    && this.rowIndex !== row.resolvedRowIndex
+                ) {
+                    this.setRowFocusState(row.resolvedRowIndex);
+                }
+            }
+        }
+    }
+
+    private focusElement(
+        element: HTMLElement,
+        focusOptions?: FocusOptions
+    ): void {
+        const previousActiveElement = this.getActiveElement();
+        if (previousActiveElement !== element) {
+            this.setElementFocusable(element, true);
+            this.isCurrentlyFocusingElement = true;
+            element.focus(focusOptions);
+            this.isCurrentlyFocusingElement = false;
+            if (
+                previousActiveElement
+                && this.isInTable(previousActiveElement)
+            ) {
+                this.setElementFocusable(previousActiveElement, false);
+            }
+        }
+    }
+
+    private setElementFocusable(
+        element: HTMLElement,
+        focusable: boolean
+    ): void {
+        if (element === this.table) {
+            return;
+        }
+
+        element.tabIndex = focusable ? 0 : -1;
+    }
+
+    private setActionMenuButtonFocused(
+        menuButton: MenuButton,
+        focused: boolean
+    ): void {
+        // The action MenuButton needs to be visible in order to be focused, so this 'focused' CSS class styling
+        // handles that (see cell/styles.ts).
+        if (focused) {
+            menuButton.classList.add('focused');
+        } else {
+            menuButton.classList.remove('focused');
+        }
```

Comment:
Can we give this a more unique class name, like `cell-action-menu-focused`
Also like the tabindex concern maybe there should be a property on the table cell, like `actionMenuShowFocused`, etc. that can be bound to so the state ownership is easier to follow. Right now that just looks like an unused class in those styles. Could be part of that same tech debt

Reply:
Updated to `cell-action-menu-focused`

Added to tech debt #2211